test/Microsoft.TemplateEngine.TestHelper/PackageManager.cs

```diff
@@ -21,10 +21,16 @@ public class PackageManager : IDisposable
     {
         private const string NuGetOrgFeed = "https://api.nuget.org/v3/index.json";
         private static readonly SemaphoreSlim Semaphore = new SemaphoreSlim(1, 1);
-        private readonly string _packageLocation = TestUtils.CreateTemporaryFolder("packages");
+        private string _packageLocation = TestUtils.CreateTemporaryFolder("packages");
         private readonly ConcurrentDictionary<string, string> _installedPackages = new ConcurrentDictionary<string, string>(StringComparer.OrdinalIgnoreCase);
         private static readonly Mutex PackMutex = new Mutex(false, "TemplateEngineTestPackMutex");
 
+        public string PackageLocation
```

Comment:
I would suggest to pass it to constructor instead of make field changable. 
Changing the field may break `_installedPackages` logic.

Reply:
`PackageManager` is also the class fixture of `Microsoft.TemplateEngine.Edge.UnitTests.NuGetInstallerTests`. Adding constructor with arguments to `PackageManager` will cause the exception below.
```
System.AggregateException : One or more errors occurred. (Class fixture type 'Microsoft.TemplateEngine.TestHelper.PackageManager' had one or more unresolved constructor arguments: String dir) (The following constructor parameters did not have matching fixture data: PackageManager packageManager)
---- Class fixture type 'Microsoft.TemplateEngine.TestHelper.PackageManager' had one or more unresolved constructor arguments: String dir
---- The following constructor parameters did not have matching fixture data: PackageManager packageManager
```

Regarding to taking packageLocation as an argument to `GetNuGetPackage`, `PackageManager` doesn't know this temporary location to delete packages while disposing as class fixture.

Reply:
I think it's ok not to remove downloaded package from custom location as soon as it is documented.
Or, custom packages downloads can be tracked in the instance additionally and removed during disposal too.

Reply:
I agree making package location alterable will bring some problem. It's better to use custom location for downloading package.